docs/guides/usage.md

```diff
@@ -0,0 +1,158 @@
+# How to use the lakeFS file system
+
+This guide contains instructions and code snippets on how to use the lakeFS file system.
+
+## The lakeFS URI structure
+
+In the following subsections, we frequently make use of [lakeFS URIs](https://docs.lakefs.io/understand/model.html#lakefs-protocol-uris) in the example code.
+lakeFS URIs are strings identifying resources in a lakeFS deployment through a unique combination of repository name, lakeFS revision/ref name, and file name relative to the repository root.
+
+As an example, a URI like `repo/main/file.txt` addresses the `file.txt` file on the `main` branch in the repository named `repo`.
```

Comment:
That's actually a bit un-ergonomic API on our part: while you can use the full URL (including the `lakefs://` scheme), the same can't be used as an rpath. Do you think it would make sense to optionally allow this kind of URI as well?

Technically, the form without the scheme isn't an URI at all (the scheme part is mandatory), so we should call it differently.

> `URI = scheme ":" ["//" authority] path ["?" query] ["#" fragment]`

Reply:
I think the majority of APIs also allow full URIs with scheme (those are stripped away in `_strip_protocol`). However, handling is inconsistent as you say, and we should canonicalize this actually. (Maybe extend the parsing regex to work on `lakefs://`-prefixed paths, too?)

Reply:
That's what I had in mind, I just don't have all the edge cases in mind right now where this might cause problems. Can you maybe check what the other implementations (`s3fs`, ...) allow?

Reply:
s3fs for one parses the bucket and resource names after stripping the protocol, so maybe calling `_strip_protocol` in the `parse` routine is the way to go here.